tests/test_diag_coded_types.py

```diff
@@ -124,7 +123,7 @@ def test_encode_leading_length_info_type_unicode2string(self):
             base_type_encoding=None,
             is_highlow_byte_order_raw=None,
         )
-        state = EncodeState(coded_message=bytes([0x12]), parameter_values={})
+        state = EncodeState(coded_message=bytes([0x12]), parameter_values={}, length_keys={})
```

Comment:
why reset length_keys?

Reply:
this is not a reset, the length keys are just explicitly specified. ([removed](7c1486c12d7d0b4a1eb840588e22cd1d5165946b)]

On that note, the current code is wrong: `length_key` is defaulted by a mutable object, so we get this carnage:
```python
from odxtools.encodestate import EncodeState

es = EncodeState(b'', parameter_values={})
es.length_keys['abc'] = 123

es2 = EncodeState(b'', parameter_values={})

print(es2.length_keys)
# -> {'abc': 123}
```

to fix this, we need to make this a dataclass and initialize these members using `field()`...